influunt-api/test/execucao/GerenciadorDeEstagiosTest.java

```diff
@@ -1416,14 +1416,15 @@ private void imprimirListaEstagios(HashMap<DateTime, IntervaloGrupoSemaforico> l
     }
 
     private void avancar(GerenciadorDeEstagios gerenciadorDeEstagios, int i) {
-        i *= 10;
-        while (i-- > 0) {
+        long quantidade = i * 10L;
+        while ((quantidade--) > 0) {
```

Comment:
Não sei se ocorre o mesmo no java, mas no javascript um `quantidade--` será primeiro avaliado, depois alterado (diferentemente do `--quantidade`). Ou seja, supondo que quantidade seja 1, a comparação seria 1 > 0, depois quantidade seria alterado para 0. Isso faria o while rodar uma vez além da quantidade necessária.

Acho interessante garantir que o comportamento seja o esperado. SE for o caso, sugiro também alterar o código e colocar o `quantidade--;` no final do while, somente por questão de legibilidade.